plugins/main/public/components/common/data-source/hooks/use-data-source.ts

```diff
@@ -125,8 +131,12 @@ export function useDataSource<
       setIsLoading(false);
     })();
 
-    return () => subscription && subscription.unsubscribe();
-  }, []);
+    return () => {
+      if (subscription) {
+        subscription.unsubscribe();
+      }
+    };
+  }, [isComponentMounted, getAbortController]);
```

Comment:
**thought**: `isComponentMounted` and `getAbortController` are memoized functions and according to the source code they do not change, so they could be removed of effect dependencies.

Reply:
Unnecessary dependencies on useEffect of use-data-source are removed